### ISSUE TITLE
Empty completionEntryDetails for auto import entry with misspelling

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#23087]
From: https://github.com/Microsoft/vscode/issues/46457

**TypeScript Version**: 2.9.0-dev-20180402

**Search Terms:** 
- completionEntryDetails
- auto imports 

**Code**
For two files in a typescript project

`a.ts`:
```ts
export const abc = 123;
```

`b.ts`:
```ts

```

In `b.ts`, type `acb`. This trigger suggestions. Go to the entry for `abc`

**Bug:**
Due to how vscode's suggestion fuzzy filtering works, we show a suggestion for `abc`  However the `completionEntryDetails` request in this case returns an empty response:

```
[Trace  - 13:39:48] Sending request: completionEntryDetails (236). Response expected: yes. Current queue length: 0
Arguments: {
    "file": "/Users/matb/projects/san/b.ts",
    "line": 1,
    "offset": 2,
    "entryNames": [
        {
            "name": "Abc",
            "source": "/Users/matb/projects/san/b"
        }
    ]
}
[Trace  - 13:39:48] Response received: completionEntryDetails (236). Request took 6 ms. Success: true 
Result: []
```

**Playground Link:**  

**Related Issues:**

**Comment by andy-ms** [comment]
This is because of the `if (stringContainsCharactersInOrder...` check in `completions.ts` -- we don't actually add the completion at `acb`, you must be using a cached version. We could remove that check, but then we'd be returning all completions from all external modules unconditionally and that might be slow. @mhegazy thoughts?

**Comment by user3323** [comment]
@mjbvz 

so there is nothing you can do with that?

I just don't understand, why do we need an auto-import "feature" that does NOT insert auto-import paths?
Or to be more accurate, does it always from SECOND try. Sp, you type name of a module or directive.... find the auto-import snippet, choose it - it didn't work, delete 1 char retype - choose snippet again and it works.

For my opinion, this ridiculous crippled "feature" should be put to rights, or removed at all.

**Comment by user3323** [comment]
@andy-ms   @mjbvz 

not sure, but seems like the same case:

![trigger1](https://user-images.githubusercontent.com/24613274/39106410-284c4e20-46c4-11e8-8755-32d3d1e718a6.gif)
